src/PublicApiGenerator/ApiGenerator.cs

```diff
@@ -675,16 +673,30 @@ static void AddPropertyToTypeDeclaration(CodeTypeDeclaration typeDeclaration, Pr
                 ? new CodeTypeReference(member.PropertyType.Name)
                 : member.PropertyType.CreateCodeTypeReference(member);
 
+            var propertyName = member.Name;
             var property = new CodeMemberProperty
             {
-                Name = member.Name,
+                Name = propertyName,
                 Type = propertyType,
                 Attributes = propertyAttributes,
                 CustomAttributes = CreateCustomAttributes(member, attributeFilter),
                 HasGet = hasGet,
                 HasSet = hasSet
             };
 
+            // DefaultMemberAttribute on type gets propagated to IndexerNameAttribute
+            var defaultMemberAttribute = typeDeclarationInfo.CustomAttributes.SingleOrDefault(x =>
+                x.AttributeType.FullName == "System.Reflection.DefaultMemberAttribute");
+            var defaultMemberAttributeArgument = defaultMemberAttribute?.ConstructorArguments.SingleOrDefault();
+            var value = defaultMemberAttributeArgument?.Value as string;
+            if (!string.IsNullOrEmpty(value) && propertyName != "Item")
```

Comment:
I think this should be `if (value == propertyName)` in case there are two of these things in the same class with different names. (VB allows you to declare any number of named indexers, but only one can be the default indexer.)

Reply:
So the question is, how do you want this to render?

```vb
Public Class C
    Public ReadOnly Default Property Foo(i As Integer) As Integer
        Get
            Return i
        End Get
    End Property
    
    Public ReadOnly Property Bar(i As Integer) As Integer
        Get
            Return i
        End Get
    End Property
End Class
```

Right now because `propertyName` is not compared to the default member name, you render them both with `[IndexerName("Foo")]` which is misleading:

```
public class C
{
    [IndexerName("Foo")]
    public int this[int i] { get; }

    [IndexerName("Foo")]
    public int this[int i] { get; }
}
```



[(sharplab)](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AbEA3YaAuIAlhgD4AKArsBkWAAQDCGAhgM5tMCwAUPf/So069AEowWAEwDyAOwwBPegBEYAMxaUM+QVAgAHGFHxKAYhAgAKIvQCCnAJKz8MAOZGAlHcfO3R3gKB9ADiMPgBQZHi+JRQsvREEZEAorKSIWFJ9Knp5HqGxgpZWUK0DOJScoq6BkYm9ABCLFDW3vROLu5QXvbtvl1ZgaHhfJFB0bHxiaMpaRkjgTk1BSa8S8zsbLxAA==)

Reply:
So the question is, how do you want this to render?

```vb
Public Class C
    Public ReadOnly Default Property Foo(i As Integer) As Integer
        Get
            Return i
        End Get
    End Property
    
    Public ReadOnly Property Bar(i As Integer) As Integer
        Get
            Return i
        End Get
    End Property
End Class
```

Right now because `propertyName` is not compared to the default member name, you render them both as default indexers which is slightly misleading, but I don't know what the alternative would be:

```
public class C
{
    [IndexerName("Foo")]
    public int this[int i] { get; }

    [IndexerName("Bar")]
    public int this[int i] { get; }
}
```

[(sharplab)](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AbEA3YaAuIAlhgD4AKArsBkWAAQDCGAhgM5tMCwAUPf/So069AEowWAEwDyAOwwBPegBEYAMxaUM+QVAgAHGFHxKAYhAgAKIvQCCnAJKz8MAOZGAlHcfO3R3gKB9ADiMPgBQZHi+JRQsvREEZEAorKSIWFJ9Knp5HqGxgpZWUK0DOJScoq6BkYm9ABCLFDW3vROLu5QXvbtvl1ZgaHhfJFB0bHxiaMpaRkjgTk1BSa8S8zsbLxAA==)

Reply:
Maybe this would be the best. The next question is whether it's worth any effort. Will anyone that uses VB use a tool that shows them their API in C#-like syntax? And will they ever use a non-default named indexer?

```cs
public class C
{
    [IndexerName("Foo")]
    public int this[int i] { get; }

    public int Bar[int i] { get; } 
}
```

Reply:
So far I haven't heard anything from VB users. And if it becomes a problem we can fix it in another version. Yagni for now

Reply:
I think rendering both as default is ok for now for such an edge case I don't want to increase the complexity because I would probably need to fallback to fancy string replacement magic again. I'd rather save my energy for the method and property declaration fixes. And maybe we can there beat this edge case because we probably require then anyway fancy property naming and replacement trickery like what we have done with events.